testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_database.py

```diff
@@ -362,3 +368,404 @@ def test_copy_prebuilt_database(params_from_base_test_setup, encrypted):
+def test_db_delete_on_active_replicator_and_live_query(params_from_base_test_setup):
+    """
+        @summary:
+        1. create a db on cbl, have a sgw available
+        2. start 2 replicators, ensure one of the replicator is push_pull replicator with continues=true
+        3. if live_query_enabled, register a live query to the cbl db, otherwise, skip this step
+        4. close the cbl db
+        5. verify cbl db is closed closed successfully
+    """
+    liteserv_version = params_from_base_test_setup["liteserv_version"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_db = "db"
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+
+    if liteserv_version < "2.8.0":
+        pytest.skip('This test supports for a feature from hydrogen(2.8.0)')
+
+    # reset SGW configuration
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # 1a. create a db, and add document to the cbl db
+    db = Database(base_url)
+    db_name = "test_delete_db_" + str(time.time())
+    log_info("Creating a Database {} at test setup".format(db_name))
+    db_config = db.configure()
+    cbl_db = db.create(db_name, db_config)
+
+    num_of_docs = 100
+    channel1 = ["Replication-1"]
+    channel2 = ["Replication-2"]
+
+    db.create_bulk_docs(num_of_docs, "ch1", db=cbl_db, channels=channel1)
+    db.create_bulk_docs(num_of_docs, "ch2", db=cbl_db, channels=channel2)
+
+    # 1b. prepare SGW, create 2 users on 2 different channels and set up connections
+    username1 = "autotest"
+    username2 = "autotest2"
+    password = "password"
+
+    sg_client = MobileRestClient()
+    sg_client.create_user(sg_admin_url, sg_db, username1, password=password, channels=channel1)
+    cookie1, session_id1 = sg_client.create_session(sg_admin_url, sg_db, username1)
+
+    sg_client.create_user(sg_admin_url, sg_db, username2, password=password, channels=channel2)
+    cookie2, session_id2 = sg_client.create_session(sg_admin_url, sg_db, username2)
+
+    # 2. start 2 replicators on 2 different channels, set all continous to True
+    replicator = Replication(base_url)
+    authenticator = Authenticator(base_url)
+
+    replicator_authenticator1 = authenticator.authentication(session_id1, cookie1, authentication_type="session")
+    repl_config1 = replicator.configure(source_db=cbl_db, replicator_authenticator=replicator_authenticator1, target_url=sg_blip_url,
+                                        replication_type="push_pull", continuous=True, channels=channel1)
+    replicator_authenticator2 = authenticator.authentication(session_id2, cookie2, authentication_type="session")
+    repl_config2 = replicator.configure(source_db=cbl_db, replicator_authenticator=replicator_authenticator2, target_url=sg_blip_url,
```

Comment:
I don't think u could use configure_and_replicate as u need to call add_change_listener before replication completed, right?

Reply:
done